projects/observability/src/shared/graphql/request/handlers/explore/explore-graphql-query-handler.service.ts

```diff
@@ -78,94 +43,6 @@ export class ExploreGraphQlQueryHandlerService
     response: GraphQlExploreServerResponse,
     request: GraphQlExploreRequest
   ): GraphQlExploreResponse {
-    return {
-      total: response.total,
-      results: response.results.map(result => this.convertResultRow(result, request))
-    };
-  }
-
-  private convertResultRow(row: GraphQlExploreServerResult, request: GraphQlExploreRequest): GraphQlExploreResult {
-    return {
-      ...this.getIntervalResultFragment(row, request),
-      ...this.getSelectionResultFragment(row, this.groupByAsSpecifications(request.groupBy)),
-      ...this.getSelectionResultFragment(row, request.selections)
-    };
-  }
-
-  private getIntervalResultFragment(
-    row: GraphQlExploreServerResult,
-    request: GraphQlExploreRequest
-  ): Pick<GraphQlExploreResult, typeof GQL_EXPLORE_RESULT_INTERVAL_KEY> {
-    if (request.interval) {
-      return {
-        [GQL_EXPLORE_RESULT_INTERVAL_KEY]: this.dateCoercer.coerce(row[INTERVAL_START_QUERY_KEY])
-      };
-    }
-
-    return {};
-  }
-
-  private getSelectionResultFragment(
-    row: GraphQlExploreServerResult,
-    specifications: ExploreSpecification[]
-  ): Pick<GraphQlExploreResult, string> {
-    return Object.assign(
-      {},
-      ...specifications.map(specification => ({
-        [specification.resultAlias()]: specification.extractFromServerData(row as Dictionary<GraphQlExploreResultValue>)
-      }))
-    );
-  }
-
-  private getAnyIntervalSelections(request: GraphQlExploreRequest): GraphQlSelection[] {
-    return request.interval ? [{ path: 'intervalStart', alias: INTERVAL_START_QUERY_KEY }] : [];
+    return this.exploreGraphQlQueryBuilderService.buildResponse(response, request);
```

Comment:
Query is just the query. If you want to include response code too, then `<>RequestResponseBuilder` would have been more clear. We can change it later :)